strongbox-event-api/src/main/java/org/carlspring/strongbox/event/artifact/ArtifactEventListenerRegistry.java

```diff
@@ -15,7 +15,7 @@
  */
 @Component
 public class ArtifactEventListenerRegistry
-        extends AbstractEventListenerRegistry<ArtifactEvent>
+        extends AbstractEventListenerRegistry
```

Comment:
I'm not sure I get why the event type has to be removed from here... What am I not seeing? Would you mind explaining?

Reply:
Event type parameter removed from `AbstractEventListenerRegistry` because we can have different Event types within one concrete `EventListenerRegistry`. As you can see we have `Map<Class<?>,List<EventListener<?>>> eventListenerMap`which assumes to have different Event types mapped to list of appropriate EventListeners.